btree/btree.c

```diff
@@ -2082,16 +2101,11 @@ static void node_put(struct node_op *op, struct nd *node, struct m0_be_tx *tx)
 		 */
 		node->n_tree = NULL;
 
-		if (!node->n_be_node_valid) {
+		if (!node->n_be_node_valid && node->n_txref == 0) {
 			ndlist_tlink_del_fini(node);
 			m0_rwlock_fini(&node->n_lock);
-			op->no_addr = node->n_addr;
-			shift = node->n_type->nt_shift(node);
-			node->n_type->nt_fini(node, tx);
 			m0_free(node);
 			m0_rwlock_write_unlock(&list_lock);
-			m0_free_aligned(segaddr_addr(&op->no_addr),
-					1ULL << shift, shift);
 			/** Capture in transaction */
```

Comment:
This comment can be removed as we are not dealing with be segment node here now.